diff --git a/gen_test_targets.bzl b/gen_test_targets.bzl
index 7dca737..cdb9c34 100644
--- a/gen_test_targets.bzl
+++ b/gen_test_targets.bzl
@@ -1,3 +1,5 @@
+load("@rules_cc//cc:cc_test.bzl", "cc_test")
+
 def gen_test_targets(name, cc_srcs):
     """Generates a cc_test target for each source file.
 
@@ -8,7 +10,7 @@ def gen_test_targets(name, cc_srcs):
 
     for src in cc_srcs:
         name = src.removeprefix("test/test_").removesuffix(".c").replace("/", "_") + "_test"
-        native.cc_test(
+        cc_test(
             name = name,
             srcs = [src],
             deps = [
